test/suites/storage_volume_initial_config.sh

```diff
@@ -0,0 +1,115 @@
+  lxc storage set "${pool}" volume.zfs.block_mode=false
+
+  lxc init "${image}" c --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+  lxc init c --empty --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+
+  # > Overwrite zfs.block_mode with initial configuration in profile.
+
+  # Verify instance "initial.zfs.block_mode=true" configuration is applied.
+  lxc storage set "${pool}" volume.zfs.block_mode=false
+  lxc profile device set "${profile}" root initial.zfs.block_mode=true
+
+  lxc init "${image}" c --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "true" ]
+  lxc delete c --force
+
+  lxc init c --empty --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "true" ]
+  lxc delete c --force
+
+  # Verify profile "initial.zfs.block_mode=false" configuration is applied.
+  lxc storage set "${pool}" volume.zfs.block_mode=true
+  lxc profile device set "${profile}" root initial.zfs.block_mode=false
+
+  lxc init "${image}" c --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+  lxc init c --empty --profile "${profile}"
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+
+  # > Verify instance overwrite of initial.* configuration.
+
+  # Verify instance "initial.zfs.block_mode=true" configuration is applied.
+  lxc storage set "${pool}" volume.zfs.block_mode=false
+  lxc profile device set "${profile}" root initial.zfs.block_mode=false
+
+  lxc init "${image}" c --profile "${profile}" --device root,initial.zfs.block_mode=true
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "true" ]
+  lxc delete c --force
+
+  lxc init c --empty --profile "${profile}" --device root,initial.zfs.block_mode=true
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "true" ]
+  lxc delete c --force
+
+  # Verify instance "initial.zfs.block_mode=false" configuration is applied.
+  lxc storage set "${pool}" volume.zfs.block_mode=true
+  lxc profile device set "${profile}" root initial.zfs.block_mode=true
+
+  lxc init "${image}" c --profile "${profile}" --device root,initial.zfs.block_mode=false
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+  lxc init c --empty --profile "${profile}" --device root,initial.zfs.block_mode=false
+  [ "$(lxc storage volume get "${pool}" container/c zfs.block_mode)" = "false" ]
+  lxc delete c --force
+
+
+  # > Verify instance initial.* configuration modification.
+
+  lxc init "${image}" c --no-profiles -s "${pool}" -d root,initial.zfs.block_mode=true
+  ! lxc config device set c root initial.zfs.blocksize=512KiB || false   # NOK: Add new configuration.
```

Comment:
Could you test that `512KiB` value can be used to set the blocksize during device creation? This would confirm that it's not ZFS rejecting it but LXD refusing it to change it if it's an `initial` value.

Reply:
Thanks for pointing that out. The error is from configuration validation because LXD sets blocksize to 128KiB even if the provided value is higher. 

I'll add some tests to ensure blocksize is being correctly set when provided through the initial configuration.